pkg/controllers/scheduling/scheduling_controller.go

```diff
@@ -352,58 +352,56 @@ func (c *schedulingController) getAvailableClusters(clusterSetNames []string) ([
 	if len(clusterSetNames) == 0 {
 		return nil, nil
 	}
+	//all available clusters
+	availableClusters := map[string]*clusterapiv1.ManagedCluster{}
 
-	// filter avilable cluserset label and values
-	avilableClusterSetsLabels := map[string]sets.String{}
 	for _, name := range clusterSetNames {
 		// ignore clusterset if failed to get
 		clusterSet, err := c.clusterSetLister.Get(name)
 		if err != nil {
 			continue
 		}
-
-		// clusterset ClusterSelector type is empty or LegacyClusterSetLabel is treated as avaliable
-		selectorType := clusterSet.Spec.ClusterSelector.SelectorType
-		if len(selectorType) == 0 || selectorType == clusterapiv1beta1.LegacyClusterSetLabel {
-			if avilableClusterSetsLabels[clusterSetLabel] == nil {
-				avilableClusterSetsLabels[clusterSetLabel] = sets.NewString()
-			}
-			// store label and values
-			avilableClusterSetsLabels[clusterSetLabel].Insert(name)
-		}
-	}
-
-	// list all avilable clusters
-	avilableClusters := map[string]*clusterapiv1.ManagedCluster{}
-	for label, vals := range avilableClusterSetsLabels {
-		requirement, err := labels.NewRequirement(label, selection.In, vals.List())
-		if err != nil {
-			klog.Warning(err)
-			continue
-		}
-
-		labelSelector := labels.NewSelector().Add(*requirement)
-		clusters, err := c.clusterLister.List(labelSelector)
+		clusters, err := c.getClusterSetClusters(clusterSet)
 		if err != nil {
-			klog.Warning(err)
-			continue
+			klog.Warning("Failed to get clusterset: %v, clusters", clusterSet.Name, err)
```

Comment:
@haoqing0110 I think we should return error in the func and reflect in the condition?

Reply:
@qiujian16 yes, PR https://github.com/open-cluster-management-io/placement/pull/72 is in progress, getAvailableClusters could return error or framework.Status (in the future) to indicate different kinds of errors and reflect in the condition.